Tryouts/Core/BasicModels/Modules/ProcessDockLayoutBehavior.cs

```diff
@@ -0,0 +1,234 @@
+﻿/// ********************************************************************************************************
+///
+/// Morgan Stanley makes this available to you under the Apache License, Version 2.0 (the "License").
+/// You may obtain a copy of the License at http://www.apache.org/licenses/LICENSE-2.0.
+/// See the NOTICE file distributed with this work for additional information regarding copyright ownership.
+/// Unless required by applicable law or agreed to in writing, software distributed under the License
+/// is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+/// See the License for the specific language governing permissions and limitations under the License.
+/// 
+/// ********************************************************************************************************
+
+using NP.Avalonia.UniDockService;
+using NP.Concepts.Behaviors;
+using System;
+using System.Reactive.Linq;
+
+namespace MorganStanley.ComposeUI.Tryouts.Core.BasicModels.Modules
```

Comment:
I'm getting confused by this namespace: do we have models, modules, view models, DTOs, or UI behaviors here?

Reply:
BasicModels is actually not for DTOs - but for the re-usable ViewModels required by the framework. In general, on the UI side it is a good practice to avoid DTOs (plain models) as much as possible dealing straight with ViewModels (that contain also DTO information). In the case above (which I now changed to composition) ModuleManifest is a Model (DTO), while ModuleViewModel is the view model, but in many cases, View Models do not have to have a separate Model counterpart. 

Modules at the end specify that these View Models are dealing with the Module functionality. Yes, unfortunately, Module  and Model are kind of similar in spelling an pronunciation, but there is no real similarity between them.